src/shared/sam/cli/samCliVersionValidator.ts

```diff
@@ -0,0 +1,143 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+import { samAboutInstallUrl } from '../../constants'
+import {
+    DefaultSamCliVersionProvider,
+    SamCliVersion,
+    SamCliVersionProvider,
+    SamCliVersionValidation
+} from './samCliVersion'
+
+const localize = nls.loadMessageBundle()
+
+export interface SamCliVersionValidatorResult {
+    version: string | undefined,
+    validation: SamCliVersionValidation
+}
+
+export class SamCliVersionValidator {
+    private static readonly ACTION_GO_TO_SAM_CLI_PAGE = localize(
+        'AWS.samcli.userChoice.visit.install.url',
+        'Get SAM CLI'
+    )
+
+    private static readonly ACTION_GO_TO_AWS_TOOLKIT_PAGE = localize(
+        'AWS.samcli.userChoice.visit.awstoolkit.url',
+        'Get AWS Toolkit'
+    )
+
+    private readonly _versionProvider: SamCliVersionProvider
+
+    public constructor(versionProvider?: SamCliVersionProvider) {
+        this._versionProvider = versionProvider || new DefaultSamCliVersionProvider()
+    }
+
+    public async validate(): Promise<SamCliVersionValidatorResult> {
+        const version = await this._versionProvider.getSamCliVersion()
+
+        return {
+            version: version,
+            validation: SamCliVersion.validate(version)
+        }
+    }
+
+    /**
+     * Call this to determine the SAM CLI version, and inform the user whether or not the version is acceptable
+     */
+    public async validateAndNotify(): Promise<void> {
+        const validationResult = await this.validate()
+
+        if (validationResult.validation === SamCliVersionValidation.Valid) {
+            await this.notifyVersionIsValid(validationResult.version!)
+        } else {
+            await this.notifyVersionIsNotValid(validationResult)
+        }
+    }
+
+    private async notifyVersionIsValid(version: string): Promise<void> {
+        vscode.window.setStatusBarMessage(
+            localize(
+                'AWS.samcli.notification.version.valid',
+                'Your SAM CLI version {0} is valid.',
+                version
+            ),
+            3333
```

Comment:
Curiously, what is the magic number?

Reply:
Updated with a named constant -- its the millis that the status bar message appears before disappearing